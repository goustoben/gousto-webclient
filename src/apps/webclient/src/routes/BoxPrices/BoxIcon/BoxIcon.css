@value ScreenMDMin from 'styles/screenSizes.css';

.container {
  position: relative;
}

.icon {
  position: absolute;
  width: 80px;
  height: 80px;
  right: -55px;
  bottom: 20px;

  @media ScreenMDMin {
    right: -70px;
    bottom: 40px;
  }
}
