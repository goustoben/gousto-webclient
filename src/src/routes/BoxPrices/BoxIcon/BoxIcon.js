import PropTypes from 'prop-types'
import React from 'react'

import { icons } from 'routes/BoxPrices/boxPricesConfig'
import css from './BoxIcon.module.css'

class BoxIcon extends React.PureComponent {
  render() {
    const { numPersons, numPortions } = this.props
    const icon = icons.find(
      (configIcon) => configIcon.numPersons === numPersons && configIcon.numPortions === numPortions
    )

    return icon && icon.image ? (
      <div className={css.container}>
        <img className={css.icon} src={icon.image} alt={icon.alt} />
      </div>
    ) : null
  }
}

BoxIcon.propTypes = {
  numPersons: PropTypes.number.isRequired,
  numPortions: PropTypes.number.isRequired,
}

export { BoxIcon }
