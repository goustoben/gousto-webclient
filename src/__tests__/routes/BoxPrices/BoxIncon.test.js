import React from 'react'
import renderer from 'react-test-renderer'

import BoxIcon from 'routes/BoxPrices/BoxIcon'

describe("BoxIcon", () => {

  it ("Will snap BoxIcon", () => {
    const tree = renderer
      .create(<BoxIcon
        numPortions={2}
        numPersons={2}
      />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
