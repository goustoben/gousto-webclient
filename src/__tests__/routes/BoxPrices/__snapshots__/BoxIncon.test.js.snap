// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoxIcon Will snap BoxIcon 1`] = `null`;
